src/transformers/models/ibert/utils/quant_modules.py

```diff
@@ -0,0 +1,538 @@
+        self.norm_type = embedding.norm_type
+        self.scale_grad_by_freq = embedding.scale_grad_by_freq
+        self.sparse = embedding.sparse
+        self.weight = embedding.weight
+
+        self.register_buffer('weight_scaling_factor', torch.zeros(1))
+        self.register_buffer('weight_integer', torch.zeros_like(self.weight))
+
+    def forward(self, x, positions=None, incremental_state=None):
+        if not self.quant_mode:
+            return F.embedding(
+                x,
+                self.weight,
+                self.padding_idx,
+                self.max_norm,
+                self.norm_type,
+                self.scale_grad_by_freq,
+                self.sparse,
+            ), None
+
+        w = self.weight
+        w_transform = w.data.detach()
+        w_min = w_transform.min().expand(1)
+        w_max = w_transform.max().expand(1)
+
+        self.weight_scaling_factor = symmetric_linear_quantization_params(
+                    self.weight_bit, w_min, w_max, False)
+        self.weight_integer = self.weight_function(
+                    self.weight, self.weight_bit, self.percentile_mode, 
+                    self.weight_scaling_factor)
+
+        emb_int = F.embedding(
+            x,
+            self.weight_integer,
+            self.padding_idx,
+            self.max_norm,
+            self.norm_type,
+            self.scale_grad_by_freq,
+            self.sparse,
+        )
+        return emb_int * self.weight_scaling_factor, self.weight_scaling_factor
+
+
+class QuantAct(Module):
+    """
+    Class to quantize given activations
+
+    Parameters:
+    ----------
+    activation_bit : int
+        Bitwidth for quantized activations.
+    act_range_momentum : float, default 0.95
+        Momentum for updating the activation quantization range.
+    per_channel : bool, default False
+        Whether to use channel-wise quantization.
+    channel_len : int, default None
+        Specify the channel length when using the per_channel mode.
+    quant_mode : bool, default False
+        The mode for quantization. True for quantization.
+    """
+    def __init__(self,
+                 activation_bit,
+                 act_range_momentum=0.95,
+                 per_channel=False,
+                 channel_len=None,
+                 quant_mode=False):
+        super(QuantAct, self).__init__()
+
+        self.activation_bit = activation_bit
+        self.act_range_momentum = act_range_momentum
+        self.quant_mode = quant_mode
+        self.per_channel = per_channel
+        self.percentile = False
+
+        if not self.per_channel:
+            self.register_buffer('x_min', torch.zeros(1))
+            self.register_buffer('x_max', torch.zeros(1))
+            self.register_buffer('act_scaling_factor', torch.zeros(1))
+        else:
+            raise NotImplementedError("per-channel mode is not currently supported for activation.")
+
+        if not self.quant_mode:
+            self.act_function = None
```

Comment:
Don't think we need to set the function to `None` if not called. I'd only add this line:
```
self.act_function = SymmetricQuantFunction.apply
```